### ISSUE TITLE
Limits of polylogarithms

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2121]
Imported from SourceForge on 2024-07-05 18:43:53
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2023-03-01 12:29:10
Original: https://sourceforge.net/p/maxima/bugs/4109

---

Maxima has code for computing limits of polylogarithms toward infinity, but the code has bugs that make some limits needlessly fail;  this one is OK

~~~
(%i1)	limit(li[2](x)/log(-x)^2,x,inf);
(%o1)	-1/2
~~~

But for order 3, we get a nounform
~~~
(%i2)	limit(li[3](x)/log(-x)^3,x,inf);
(%o2)	limit(li[3](x)/log(-x)^3,x,inf)
~~~

The nounform is due to  calling `li-asymptotic-expansion` with arguments that violate the requirement `pw < s/2.`  The result is that  `gamma` is called  on a negative integer in a denominator.   Actually, I think that all such terms are supposed to be zero.

Another problem is that I think the code has a 2 that should be `s.'
~~~
(defun li-asymptotic-expansion (pw s z)
    (m+l (loop for k from 0 to pw collect
	     (m* (m^ -1 k)
		 (m- 1 (m^ 2 (m- 1 (m* 2 k))))
		 (m^ (m* 2 '$%pi) (m* 2 k))
		 (m// ($bern (m* 2 k))
		      `((mfactorial) ,(m* 2 k)))
		 (m// (m^ `((%log) ,(m- z)) (m- 2 (m* 2 k))) ; should be (m- s (m* 2 k))?
		      ($gamma (m+ s 1 (m* -2 k))))))))
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:43:54
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2023-03-01 13:48:52
Original: https://sourceforge.net/p/maxima/bugs/4109/#aa73

---

One more thing:  possibly, the function `li-asymptotic-expansion` should set `zerobern` to true (the default).

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:43:58
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2023-05-12 18:23:40
Original: https://sourceforge.net/p/maxima/bugs/4109/#39d9

---

thanks for the suggestions, fixed by [bef00a3913f06cd7c914356b68e4ec7faceafb6e]

(did not set zerobern to true - feel free to add this)

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:44:01
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2023-05-12 18:23:58
Original: https://sourceforge.net/p/maxima/bugs/4109/#3661

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea
